clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -744,16 +767,34 @@ public boolean isLeavingGroup() {
         return state == MemberState.PREPARE_LEAVING || state == MemberState.LEAVING;
     }
 
+    @Override
+    public void maybeRejoinStaleMember() {
+        if (state == MemberState.STALE) {
+            staleMemberAssignmentRelease.whenComplete((__, error) -> transitionToJoining());
```

Comment:
Can we avoid enqueuing many of these completionstages while we haven't finished the `staleMemberAssignmentRelease`? although it looks like the operation is idempotent, it's not good to create an arbitrary number of Futures here (with every `PollApplicationEvent`, a new one is created!

Reply:
Sure, good catch, done. We truly only need it when an expired poll timer is reset. Note that I added the check in the HB (where I'm trying to keep the poll timer logic), and the membershipMgr is only concern about the stale situation (which is more than just the timer concept, because it includes the callbacks that may take a while)